logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -65,6 +65,9 @@ public void seekToNextEvent(Timestamp timestamp) throws IOException {
             currentReader = new RecordIOReader(segment);
             byte[] event = currentReader.seekToNextEventPosition(timestamp, (b) -> {
                 try {
+                    if (b == null){
```

Comment:
since you added the null->break check for `readEvent` in the RecordIOReader, it can now be assumed that this keyExtractor will only be called if `b != null`. so this check is no longer needed, right?